crates/state/src/client_state.rs

```diff
@@ -115,6 +115,13 @@ impl ClientState {
     }
 }
 
+#[cfg(feature = "test_utils")]
+impl ClientState {
+    pub fn set_last_finalized_checkpoint(&mut self, chp: L1Checkpoint) {
+        self.local_l1_view.last_finalized_checkpoint = Some(chp);
+    }
```

Comment:
Given how this is used, shouldn't this be `#[cfg(test)]`?

Reply:
The thing is, this cannot be done because this method is used in unit test of another crate, and using the feature flag is the only way. I had the same assumption and spent a whole lot of time trying to do it with `#[cfg(test)]`

Reply:
Oh that's unfortunate.